binder/src/main/java/io/grpc/binder/AndroidComponentAddress.java

```diff
@@ -41,7 +40,6 @@
  * fields, namely, an action of {@link ApiConstants#ACTION_BIND}, an empty category set and null
  * type and data URI.
  */
-@ExperimentalApi("https://github.com/grpc/grpc-java/issues/8022")
 public class AndroidComponentAddress extends SocketAddress { // NOTE: Only temporarily non-final.
   private static final long serialVersionUID = 0L;
 
```

Comment:
At the checkArgument call below (line 49), I suspect we should add "|| bindIntent.getPackage() != null"

Reply:
Added an additional argument check for package. Can remove if @jdcormie has other thoughts.

Reply:
@jdcormie , wanted to follow up on this.

Reply:
I agree it would be nice to allow this but I think there's a lot more to it than just relaxing the check here. See CR/446742665 or CR/451425849 for what I think it would take.

Reply:
I'm not entirely sure that relaxing the API is a good change generally speaking, and I think it would be better to leave as strictly a component name check for now.

Could this change be revisited in a later PR?

Reply:
sgtm.